File: project/Full_Stack_Assignment/frontend/src/services/CartService.js
import axios from "axios";

const BASE_URL = "http://localhost:8082/api/cart";

export const getCartItems = (userId) => {
  return axios.get(`${BASE_URL}/${userId}/items`);
};

export const addItemToCart = (
  userId,
  productId,
  quantity,
  price,
  productName
) => {
  return axios.post(`${BASE_URL}/add`, {
    userId,
    productId,
    quantity,
    price,
    productName,
  });
};

export const removeCartItem = (cartItemId) => {
  return axios.delete(`${BASE_URL}/item/${cartItemId}`);
};

export const clearCart = (userId) => {
  return axios.delete(`${BASE_URL}/${userId}/clear`);
};

export const checkoutCart = (userId, { address }) => {
  return axios.post(`${BASE_URL}/${userId}/checkout`, { address });
};

File: project/Full_Stack_Assignment/backend/orders-service/src/main/java/com/orders/carts/service/CartService.java
package com.orders.carts.service;

import java.math.BigDecimal;
import java.util.List;

import com.orders.carts.domain.Cart;
import com.orders.carts.domain.CartItem;

public interface CartService {
    Cart getOrCreateCart(Long userId);
    void addItemToCart(Long userId, Long productId, Integer quantity, BigDecimal price,String productName);
    List<CartItem> getCartItems(Long userId);
    void removeItem(Long cartItemId);
    void clearCart(Long userId);
	void checkout(Long userId, String address);
}
